tests/async.rs

```diff
@@ -0,0 +1,70 @@
+#![cfg(feature="unstable")]
+
+extern crate futures;
+extern crate tokio_core;
+extern crate tokio_io;
+extern crate reqwest;
+extern crate libflate;
+
+#[macro_use]
+mod support;
+
+use futures::{Future, Stream};
+use tokio_core::reactor::Core;
+use std::io::Write;
+use std::time::Duration;
+
+#[test]
+fn async_test_gzip_response() {
+    let content: String = (0..10000).into_iter().map(|i| format!("test {}", i)).collect();
+    let chunk_size = content.len() / 10;
+    let mut encoder = ::libflate::gzip::Encoder::new(Vec::new()).unwrap();
+    match encoder.write(content.as_bytes()) {
+        Ok(n) => assert!(n > 0, "Failed to write to encoder."),
+        _ => panic!("Failed to gzip encode string."),
+    };
+
+    let gzipped_content = encoder.finish().into_result().unwrap();
+
+    let mut response = format!("\
+            HTTP/1.1 200 OK\r\n\
+            Server: test-accept\r\n\
+            Content-Encoding: gzip\r\n\
+            Transfer-Encoding: chunked\r\n\
+            Content-Length: {}\r\n\
+            \r\n", &gzipped_content.len())
+        .into_bytes();
+    response.extend(&gzipped_content);
+
+    let server = server! {
+        request: b"\
+            GET /gzip HTTP/1.1\r\n\
+            Host: $HOST\r\n\
+            User-Agent: $USERAGENT\r\n\
+            Accept: */*\r\n\
+            Accept-Encoding: gzip\r\n\
+            \r\n\
+            ",
+        chunk_size: chunk_size,
+        write_timeout: Duration::from_secs(1),
```

Comment:
I recommend a smaller delay here. 5~10 ms is enough for machines today, as the operation happens at microseconds.

Also, I would recommend an extreme case with chunked encoding of 1 byte each line. That stress tests the buffer implementation to provide a correct implementation of read_exact.